cmd/soroban-rpc/internal/db/db.go

```diff
@@ -366,3 +373,14 @@ func runSQLMigrations(db *sql.DB, dialect string) error {
 	_, err := migrate.ExecMax(db, dialect, m, migrate.Up, 0)
 	return err
 }
+
+func NewTestDB(tb testing.TB) *DB {
+	tmp := tb.TempDir()
+	dbPath := path.Join(tmp, "db.sqlite")
+	db, err := OpenSQLiteDB(dbPath)
+	require.NoError(tb, err)
+	tb.Cleanup(func() {
+		assert.NoError(tb, db.Close())
+	})
+	return db
+}
```

Comment:
I am not sure why this is needed, but it shouldn't live with production code (it causes bloat). Consider placing it in a `_test.go` file or a package only used by tests.